policy-controller/grpc/src/outbound.rs

```diff
@@ -222,18 +252,138 @@ fn response_stream(
     })
 }
 
-fn to_service(
-    outbound: OutboundPolicy,
+fn external_stream(
+    drain: drain::Watch,
+    mut rx: ExternalPolicyStream,
+    original_dst: SocketAddr,
+) -> BoxWatchStream {
+    Box::pin(async_stream::try_stream! {
+        tokio::pin! {
+            let shutdown = drain.signaled();
+        }
+
+        loop {
+            tokio::select! {
+                res = rx.next() => match res {
+                    Some(_) => {
+                        yield fallback(original_dst);
+                    }
+                    None => return,
+                },
+
+                // If the server starts shutting down, close the stream so that it doesn't hold the
+                // server open.
+                _ = &mut shutdown => {
+                    return;
+                }
+            }
+        }
+    })
+}
+
+fn fallback(original_dst: SocketAddr) -> outbound::OutboundPolicy {
+    // This encoder sets deprecated timeouts for older proxies.
+    #![allow(deprecated)]
```

Comment:
I think this can be removed if we use `..Default::default()` to fill in default values for request_timeout instead of setting to None explicitly.